### ISSUE TITLE
Commenting system

### ISSUE THREAD

**Issue by saurabhd** [snehal-addweb/Mortgagespeak#41]
We should show "Comments" on the "details" page for news, press releases, perspectives, etc. We don't current show a details page for "News" but for commenting, it may make sense for this functionality only.

Sample image attached
![commenting-main-table-display](https://cloud.githubusercontent.com/assets/2010385/14711658/e2a83ed6-07f7-11e6-853f-e65663b2fc03.png)
![comment-details-page-with-buttons](https://cloud.githubusercontent.com/assets/2010385/14711659/e2ad96ce-07f7-11e6-9a89-6302f93a821c.png)

**Comment by saurabhd** [comment]
ball park estimation for this task is 8 -10 hours.

**Comment by gatsby8870** [comment]
This estimate is fine. Thank you. 

Does this include replying to a user's comments as well?

Attached is the same image with a minor edit to the arrow of the "bubble".

![comment-details-page-with-buttons](https://cloud.githubusercontent.com/assets/18587752/14726775/6707ad28-07ea-11e6-87d7-fcdd588113f0.png)

**Comment by saurabhd** [comment]
Yes, it includes the replying, but not vote up and down, which has been shown in your graphics. Drupal provides the default commenting system and we are going to use it, in terms of look and feel, Ill try to make it as close as possible in the given time frame, but if you have any specific requirement, please state here, so that I can evaluate it.

**Comment by gatsby8870** [comment]
That sounds fine. I am looking into Drupal thumbs up/down modules. We can always add it later on perhaps.

**Comment by gatsby8870** [comment]
I am hoping this is done in the next day or so. I have a tour this Friday and this is a key part. Looks like it will but wanted to share.

Let me know if I can help.

**Comment by gatsby8870** [comment]
Hello, is there an update on this you can provide?

Thank you.

**Comment by saurabhd** [comment]
Hello Greg,

Here's the updates:
1. Public user will be able to post comment on content, but admin have to approve that comments.
2. For Public User
   => "Your Name" and "Comment" textbox will be displayed.
3. For Logged In (i.e NI) User
   => Only "Comment" textbox will be displayed.

Note : You have to approve "Public users" comments from (http://mortgage.saurabhdhariwal.com/admin/content/comment/approval) or from that particular content's detail page

QA found some issue, working on those.

**Comment by Krina-Addweb** [comment]
1) It should display anonymous username for Comment which is missed right now & also Remove Gap in front from the timing when we comment.

**Comment by gatsby8870** [comment]
Hello, this project is marked as complete but there are two important items still missing.

1). In this thread, there are two images supplied. The first one is a web page display of all the content. Beneath each headline and summary gray text are social networking icons. The last icon is a green "Comment" icon. This icon and "Add comment" hyperlink are displayed when there is no comments made. When comments are made, it displays the number of comments and is hyperlinked to comment functionality.

2). On the content details page (where the functionality is), there should be a green "Comment" button in the same row as the existing social icons: Twitter, Facebook, Google +, etc. The second image in this thread at the top shows this display.

Please let me know how I may help! Thank you.

**Comment by gatsby8870** [comment]
Regarding #2, the green "Comment" button can simply scroll the web page down to the commenting functionality.

Thank you.

**Comment by saurabhd** [comment]
Hi Greg,

I think you need to clarify the flow here, we do not have the detail pages where the link is external, so there is no point in adding comment system to it. As I mentioned I have estimated the task based on the scenario provided by the drupal defaults. 

Please provide your views, so that I can decide my approach.

Thanks
Saurabh

**Comment by gatsby8870** [comment]
Let me consider the "News" details page today. My initial thought was clicking the "Comment" link for that content would open a (new) details page (pulling summary gray text) and the the user could comment below. I will let you know regarding this. Likely not to do it at this time.

However, the green "Comment" icon and text (and button) displayed in the main table, pop up, and details page should appear for the existing details pages now:
1. Press Releases
2. Perspectives

**Comment by saurabhd** [comment]
I think, a concrete instruction will help to have a better plan for doing this and avoid rework. 
will wait for your complete task and then will proceed, I am working on this Saturday and planning to use all my hours in this project.

Thanks

**Comment by gatsby8870** [comment]
Ok.

**As an update, just add the existing comment functionality you did to the "Perspectives" details pages and hold all items.**

I am not authorizing all day Sat work and I would like this done by my time tomorrow morning as I have a tour Friday as mentioned previously.

Thank you and let me know of any issues.
